scripts/scil_compute_ihMT_maps.py

```diff
@@ -0,0 +1,442 @@
+                        Altnp = 0; Atlpn = 1; Reference = 2; Negative = 3;
+                        Positive = 4; T1weighted = 5
+    Returns
+    ----------
+    Thresholded matrix in 3D-array.
+    """
+    # Remove NaN and apply thresold based on lower and upper value
+    computed_map[np.isnan(computed_map)] = 0
+    computed_map[np.isinf(computed_map)] = 0
+    computed_map[computed_map < lower_threshold] = 0
+    computed_map[computed_map > upper_threshold] = 0
+
+    # Load and apply sum of T1 probability maps on myelin maps
+    mask_image = nib.load(in_mask)
+    mask_data = get_data_as_mask(mask_image)
+    computed_map[np.where(mask_data == 0)] = 0
+
+    # Apply threshold based on combination of specific contrasts maps
+    for idx in idx_contrast_list:
+        computed_map[contrast_maps[idx] == 0] = 0
+
+    return computed_map
+
+
+def main():
+    parser = _build_arg_parser()
+    args = parser.parse_args()
+
+    assert_output_dirs_exist_and_empty(parser, args,
+                                       os.path.join(args.out_dir,
+                                                    'Contrasts_ihMT_maps'),
+                                       os.path.join(args.out_dir,
+                                                    'ihMT_native_maps'),
+                                       create_dir=True)
+
+    # Merge all echos path into a list
+    maps = [args.in_altnp, args.in_altpn, args.in_mtoff, args.in_negative,
+            args.in_positive, args.in_t1w]
+
+    maps_flat = (args.in_altnp + args.in_altpn + args.in_mtoff +
+                 args.in_negative + args.in_positive + args.in_t1w)
+
+    jsons = [curr_map.replace('.nii.gz', '.json')
+             for curr_map in maps_flat]
+
+    # check data
+    assert_inputs_exist(parser, jsons + maps_flat)
+    for curr_map in maps[1:]:
+        if len(curr_map) != len(maps[0]):
+            parser.error('Not the same number of echoes per contrast')
+
+    # Set TR and FlipAngle parameters for ihMT (positive contrast)
+    # and T1w images
+    parameters = [set_acq_parameters(maps[4][0].replace('.nii.gz', '.json')),
+                  set_acq_parameters(maps[5][0].replace('.nii.gz', '.json'))]
+
+    # Fix issue from the presence of NaN into array
+    np.seterr(divide='ignore', invalid='ignore')
+
+    # Define reference image for saving maps
+    ref_img = nib.load(maps[4][0])
+
+    # Define contrasts maps names
+    contrasts_name = ['altnp', 'altpn', 'reference', 'negative', 'positive',
+                      'T1w']
+    if args.filtering:
+        contrasts_name = [curr_name + '_filter'
+                          for curr_name in contrasts_name]
+
+    # Compute contrasts maps
+    computed_contrasts = []
+    for idx, curr_map in enumerate(maps):
+        computed_contrasts.append(compute_contrasts_maps(
+                                  curr_map, filtering=args.filtering))
+
+        nib.save(nib.Nifti1Image(computed_contrasts[idx].astype(np.float64),
```

Comment:
Do we really need np.float64 ? Unless the value are known to be below 1x10^6 and lower we go for np.float32